homeassistant/components/unifiprotect/sensor.py

```diff
@@ -771,20 +771,22 @@ def __init__(
     def _async_update_device_from_protect(self, device: ProtectModelWithId) -> None:
         # do not call ProtectDeviceSensor method since we want event to get value here
         EventEntityMixin._async_update_device_from_protect(self, device)
-        if (
-            self.entity_description.ufp_smart_type
-            == SmartDetectObjectType.LICENSE_PLATE
-        ):
+        is_on = self.entity_description.get_is_on(device)
+        if is_on and self._event is not None:
             if (
-                self._event is None
-                or self._event.metadata is None
-                or self._event.metadata.license_plate is None
+                self.entity_description.ufp_smart_type
+                == SmartDetectObjectType.LICENSE_PLATE
             ):
-                self._attr_native_value = OBJECT_TYPE_NONE
-            else:
-                self._attr_native_value = self._event.metadata.license_plate.name
-        else:
-            if self._event is None:
-                self._attr_native_value = OBJECT_TYPE_NONE
+                if (
+                    self._event.metadata is not None
+                    and self._event.metadata.license_plate is not None
+                ):
+                    self._attr_native_value = self._event.metadata.license_plate.name
+                else:
+                    self._attr_native_value = OBJECT_TYPE_NONE  # pragma: no cover
             else:
                 self._attr_native_value = self._event.smart_detect_types[0].value
+        else:
```

Comment:
You could make the code more readable by converting it to guard clauses and saving on indentation.

```python
if not is_on or self._event is None:
    self._attr_native_value = OBJECT_TYPE_NONE
    self._event = None
    self._attr_extra_state_attributes = {}
    return
```

Reply:
I cleaned it up a bit and got rid of the pragma, but it still does not look the best. 

It is only temporary though as once the other sensor is removed in 2023.3.0, I can clean up the method.